broker/redis/redis.go

```
@@ -0,0 +1,240 @@
+package redis
+
+import (
+	"encoding/json"
+	"errors"
+	"strings"
+	"time"
+
+	"golang.org/x/net/context"
+
+	"github.com/garyburd/redigo/redis"
+	"github.com/micro/go-micro/broker"
+	"github.com/micro/go-micro/cmd"
+)
+
+func init() {
+	cmd.DefaultBrokers["redis"] = NewBroker
+}
+
+type publication struct {
+	topic   string
+	message *broker.Message
+}
+
+func (p *publication) Topic() string {
+	return p.topic
+}
+
+func (p *publication) Message() *broker.Message {
+	return p.message
+}
+
+func (p *publication) Ack() error {
+	return nil
+}
+
+type subscriber struct {
+	conn   *redis.PubSubConn
+	topic  string
+	handle broker.Handler
+	opts   broker.SubscribeOptions
+}
+
+func (s *subscriber) recv() {
+	// Close the connection once the subscriber stops receiving.
+	defer s.conn.Close()
+
+	for {
+		switch x := s.conn.Receive().(type) {
+		case redis.Message:
+			var m broker.Message
+
+			// Handle error?
+			if err := json.Unmarshal(x.Data, &m); err != nil {
```

I see what you mean here. Basically about swallowing errors. Some return an error channel for something like this. Could augment the subscriber to add something of that nature.